core/src/validator.rs

```diff
@@ -168,7 +170,44 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bf = bank_forks.clone();
+        let kp = keypair.clone();
+        let mut counter = 0;
+        let counter_ref = &mut counter;
+        if counter == 0 {
+            set_handler(move || { 
+                *counter_ref += 1;
+                let bank = bf.read().unwrap().working_bank();
+                let validator_pubkey = kp.pubkey();
+                let mut current_stakes: Vec<_>  = bank
+                .epoch_vote_accounts(bank.get_epoch_and_slot_index(bank.slot()).0)
+                .map(|accounts| {
+                    accounts
+                        .iter()
+                        .filter_map(|(pubkey, (stake, account))| {
+                            if let Ok(vote_state) = VoteState::deserialize(&account.data) {
+                                if vote_state.node_pubkey == validator_pubkey { 
+                                    Some((*pubkey, *stake))
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        })
+                        .collect()
+                })
+                .unwrap();
+                if current_stakes.len() > 0 {
```

Comment:
Hmm, I think we could just move the `warn!()` into the `.filter_map()` and then remove this entire block.